### PR TITLE
Skip prompting the user if entity or table file is unchanged

### PR DESCRIPTION
**Pull request by othercorey**
Refs https://github.com/cakephp/bake/issues/709

### CODE REVIEW

File: src/Command/BakeCommand.php [hunk 1]
```diff
@@ -196,4 +196,32 @@ protected function parseFile(string $path): ?ParsedFile
...
+     * @return bool True if successful, false otherwise
+     * @throws \Cake\Console\Exception\StopException When `q` is given as an answer
+     *   to whether a file should be overwritten.
+     */
+    protected function writeFile(
+        ConsoleIo $io,
+        string $path,
+        string $contents,
+        bool $forceOverwrite = false,
+        bool $skipSame = true,
```
Comment (othercorey) [inline src/Command/BakeCommand.php]: This could be an option to ConsoleIo::createFile()

**Comment by othercorey** [comment]
We might need to support ignoring newline differences to support re-gen on windows with \n\r but the project saving as \n. This is currently why the windows test fails.

File: src/Command/BakeCommand.php [hunk 2]
```diff
@@ -196,4 +196,32 @@ protected function parseFile(string $path): ?ParsedFile
...
+     * @return bool True if successful, false otherwise
+     * @throws \Cake\Console\Exception\StopException When `q` is given as an answer
+     *   to whether a file should be overwritten.
+     */
+    protected function writeFile(
+        ConsoleIo $io,
+        string $path,
+        string $contents,
+        bool $forceOverwrite = false,
+        bool $skipSame = true
```
Comment (ADmad) [inline src/Command/BakeCommand.php]:
```suggestion
        bool $skipIfSame = true
```
Sounds better

Comment (othercorey) [reply src/Command/BakeCommand.php]: I wasn't sure, but seeing that, I wonder if `skipIfUnchanged` is clearer. I used that name in the tests.

Comment (ADmad) [reply src/Command/BakeCommand.php]: `skipIfUnchanged` 👍

File: tests/TestCase/Command/ModelCommandTest.php [hunk 3]
```diff
@@ -1659,6 +1659,26 @@ public function testBakeEntitySimple()
...
+    /**
+     * Tests baking a file with no changes
+     *
+     * @return void
+     */
+    public function testBakeEntitySimpleUnchanged(): void
+    {
+        $this->generatedFile = APP . 'Model/Entity/User.php';
+        $result = file_get_contents($this->_compareBasePath . __FUNCTION__ . '.php');
+        file_put_contents($this->generatedFile, str_replace("\r\n", "\n", $result));
```
Comment (othercorey) [inline tests/TestCase/Command/ModelCommandTest.php:1671]: This forces the newlines to match on windows.

File: src/Command/BakeCommand.php [hunk 4]
```diff
@@ -212,4 +212,32 @@ protected function parseFile(string $path): ?ParsedFile
...
+     */
+    protected function writeFile(
+        ConsoleIo $io,
+        string $path,
+        string $contents,
+        bool $forceOverwrite = false,
+        bool $skipIfUnchnged = true
+    ): bool {
+        if ($skipIfUnchnged && file_exists($path) && file_get_contents($path) === $contents) {
+            $io->info("{$path} already exists but has no changes, skipping");
```
Comment (markstory) [inline src/Command/BakeCommand.php]:
```suggestion
            $io->info("Skipping update to `{$path}`. It already exists and would not change.");
```

File: tests/TestCase/Command/ModelCommandTest.php [hunk 5]
```diff
@@ -1659,6 +1659,26 @@ public function testBakeEntitySimple()
...
+
+        $this->exec('bake model --no-test --no-fixture --no-table --no-fields --no-hidden users');
+
+        $this->assertExitCode(CommandInterface::CODE_SUCCESS);
+        $this->assertFileExists($this->generatedFile);
+        $this->assertSameAsFile(__FUNCTION__ . '.php', $result);
+
+        $this->assertOutputContains("{$this->generatedFile} already exists but has no changes, skipping");
```
Comment (markstory) [inline tests/TestCase/Command/ModelCommandTest.php]:
```suggestion
        $this->assertOutputContains("Skipping update to `{$this->path}`. It already exists and would not change.");
```

**Pull request merged**